api/modules/publishedFiles/model.js

```diff
@@ -1,9 +1,101 @@
 var BaseModel = require('../../classes/base_model');
 
+var File = require('../files/model');
+
 var instanceProps = {
   tableName: 'publishedFiles',
   project: function() {
     return this.belongsTo(require('../publishedProjects/model'));
+  },
+  queries: function() {
+    var self = this;
+    var FilesTable = File.prototype;
+    var PublishedFile = this.constructor;
+
+    return {
+      getOne: function(id) {
+        return new PublishedFile().query()
+        .where(self.column('id'), id)
+        .then(function(publishedFiles) {
+          return publishedFiles[0];
+        });
+      },
+      getAllNewFiles: function(projectId) {
+        return new File().query()
+        .leftOuterJoin(self.tableName,
+                       FilesTable.column('id'),
+                       self.column('file_id'))
+        .where(FilesTable.column('project_id'), projectId)
+        .whereNull(self.column('file_id'))
```

Comment:
When you do a left outer join, it returns everything from the "left" table regardless of whether it refers to anything on the right table or not. We only want the values which do not refer to anything in the right table...so yeah...your second comment basically :P